src/frontend/sharedComponents/ApiLoading/index.tsx

```diff
@@ -24,10 +27,17 @@ export const ApiLoading = ({children}: React.PropsWithChildren<{}>) => {
 };
 
 function useNodejsMobileInit() {
+  const [rootKey, status, setRootKey] = useSecureStore('RootKey');
+
+  if (!rootKey && status !== 'loading') {
+    setRootKey(uint8ArrayToHex(getRandomBytes(16)));
+  }
+
   React.useEffect(() => {
-    // TODO: Use startWithArgs and pass in other args that the backend expects
-    nodejs.start('loader.js');
-  }, []);
+    if (!rootKey) return;
+
+    nodejs.startWithArgs(`loader.js --rootKey ${rootKey}`);
```

Comment:
ideally the start would only be called once. in theory, this could be at risk calling start multiple times. 

not sure what the proper solution should be but for now, perhaps adding a ref to indicate that it's been called already and checking it here would make sense in addition to this?

Reply:
Is there a state from the backend that we could check that indicates whether nodejsmobile has been started? That seems like a better solution. 

If that's not possible, maybe it would be better to make sure that the root key cannot be changed. Like do an explicit check any time `setRootKey` is being run, and don't set it unless the `rootKey` Value is null

Reply:
> Is there a state from the backend that we could check that indicates whether nodejsmobile has been started? That seems like a better solution.

ah yeah now that you mention it, we use the server status to initiate the nodejs mobile script:

https://github.com/digidem/mapeo-mobile/blob/56528dcccf291d4248beb6348767ca50a6ce92be/src/frontend/AppLoading.tsx#L77-L81

so perhaps that should be incorporated here

Reply:
So does [this](https://github.com/digidem/CoMapeo-mobile/pull/89/commits/47fbe4cd6d19b0dd9628cd1c0ed9271442a32ea5) make sense?